src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -324,7 +324,15 @@ private void TextBuffer_OnChanged(object sender, TextContentChangedEventArgs arg
             }
             else
             {
-                TryUpdateLatestParsedSyntaxTreeSnapshot(partialParseSyntaxTree, snapshot);
+                var currentCodeDocument = CodeDocument;
+                var codeDocument = RazorCodeDocument.Create(currentCodeDocument.Source, currentCodeDocument.Imports);
```

Comment:
Is this really right? One of the things that makes me anxious about this change is that there are too many ways to create a code document

Reply:
I definitely understand your concern, creating a code document is like starting a new character in Skyrim, except the nose, ears, head and body are optional 🤣.

That being said the current directive attribute requirements will need the the list of TagHelpers for the current document. With just that requirement we could pass that info in; however, if I ever want to add support for TagHelper completions (I already do in VSCode) I need things like the TagHelper prefix. Also, if we ever want to move TagHelpers away from the parsing phase and up into IR we'll most likely need the IR document to understand when TagHelper completions should or should not apply.

If you feel strongly that I'm being crazy here we can just pass in the `TagHelperDocumentContext`  (has TagHelpers and prefix) along side the `SyntaxTree` here and in providers and then address the future future concerns of a TagHelperless `SyntaxTree` later.

Reply:
I would feel a heck of a lot better about passing in the tag helper document context.

Reply:
related: https://www.youtube.com/watch?v=q6yHoSvrTss

Reply:
lol, ya that was worth the 3:30. This should be required homework for all of us

Reply:
Just to note, since i'll still need the TagHelper document context I still need to maintain some level of "latest code document information". This approach of copying over all items/bits to the code document should satisfy every possible manipulation of the code document because by itself the code document is pretty light. The items it contains are where things can differ wildly.